errorDashboardJS/src/apiClient.ts

```diff
@@ -0,0 +1,107 @@
+import { customFetch } from "./fetch";
+import { configs, type Configs } from "./configs";
+import { baseUrl } from "./environment";
+import type { CreateErrorRequestType, CreateTagRequestType } from "./types";
+
+interface InitializeClient {
+  clientId: string;
+  clientSecret: string;
+  window: Window;
+}
+
+export class ErrorDashboardClient {
+  private clientId: string;
+  private clientSecret: string;
+  private window: Window;
+  private configs: Configs;
+
+  constructor(obj: InitializeClient) {
+    this.clientId = obj.clientId;
```

Comment:
Sentry accepts an `undefined` DSN, which is similar to your secret/id. They just log that reporting won't happen but continue on. Worth considering.

[Code here.](https://github.com/getsentry/sentry-javascript/blob/master/packages/core/src/baseclient.ts#L133-L137)

Reply:
In general, I don't want to report errors during dev but I do want to on `stage`, `qa`, or `prod` environments.